QMLComponents/utilities/messageforwarder.cpp

```diff
@@ -30,6 +30,55 @@ bool MessageForwarder::useNativeFileDialogs()
 	return DesktopCommunicator::singleton()->useNativeFileDialog();
 }
 
+bool MessageForwarder::engineSandbox()
+{
+	return DesktopCommunicator::singleton()->engineSandbox();
+}
+
+QString MessageForwarder::constrainToSandboxResult(const QString &selectedPath, bool file, bool save)
```

Comment:
So I really think better behaviour is:

That path the user selects is what they get.
In jaspBase we add  `jaspFileWriter` or something. It simply stores whatever the R code gives it into our temp dir.
Then sends a special entry back in the results that isnt shown in the results at all, but will tell desktop about the file and lets it copy it there. It can have a reference to the option that contains the path, or the path or whatever.

I can help out with setting this up.

For now this would  be ok in `development` but if we release it like this I think its a bit too ugly.

Reply:
We should discuss this tomorrow/today a bit.
I do agree with you this is a bit too jank for your average user.